test/e2e/apps/daemon_set.go

```diff
@@ -352,90 +336,13 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to start")
 
 		// Check history and labels
-		ds, err = c.ExtensionsV1beta1().DaemonSets(ns).Get(ds.Name, metav1.GetOptions{})
+		ds, err = c.AppsV1().DaemonSets(ns).Get(ds.Name, metav1.GetOptions{})
 		Expect(err).NotTo(HaveOccurred())
 		waitForHistoryCreated(c, ns, label, 2)
 		cur = curHistory(listDaemonHistories(c, ns, label), ds)
-		hash = cur.Labels[extensions.DefaultDaemonSetUniqueLabelKey]
+		hash = cur.Labels[apps.DefaultDaemonSetUniqueLabelKey]
 		Expect(cur.Revision).To(Equal(int64(2)))
-		checkDaemonSetPodsLabels(listDaemonPods(c, ns, label), hash, fmt.Sprint(templateGeneration))
-	})
-
-	It("Should adopt existing pods when creating a RollingUpdate DaemonSet regardless of templateGeneration", func() {
```

Comment:
This tests adoption using hash (should work even when templateGeneration doesn't match). We should keep this and remove parts about templateGeneration. 

Also change the description to "Should adopt existing pods when creating a RollingUpdate DaemonSet".

Reply:
I think we should do this as an integration test. I can see the value for e2e when TemplateGeneration was a supported, user facing feature. The template hash, and its interaction with orphaning and adoption, is an implementation detail of the controller that requires only the API server and the controller manager to test.

Reply:
Moving it to integration sounds good.

Reply:
I came out with the PR to make it an integration test: #59013
I will wait until this PR is merged, change my test to use `apps/v1`, and then ask for review.